File: RedSheet.DbEntities/Models/vDefaultSetting.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vDefaultSettings", Schema = "dbo")]
	[Serializable]
	public partial class vDefaultSetting  
	{
 
	[Required]
	public decimal Cost { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int DefaultSettingId { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string DefaultSettingName { get; set; } 
 
	[Required] 
	[MaxLength(100)]
	public string Status { get; set; } 

	public vDefaultSetting()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-default-setting.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vDefaultSetting {
    constructor(vDefaultSetting?: vDefaultSetting )  {
        let properties = [ "cost", "defaultSettingId", "defaultSettingName", "status",];
        for (let property of properties)
            if (vDefaultSetting && vDefaultSetting[property])
                this[property] = vDefaultSetting[property];
    }
 
    @required()
	cost : number =   undefined;
 
	defaultSettingId : number =   0 ;
 
    @required()
    @maxLength(200)
	defaultSettingName : string =   undefined;
 
    @required()
    @maxLength(100)
	status : string =   undefined;


}
